src/main/java/dev/dirs/Util.java

```diff
@@ -214,6 +214,47 @@ private static void initBase64Encoding() {
     }
   }
 
+  private static String[] runWinCommands(int guidsLength, Charset charset, String[] dirs, String encodedCommand) throws IOException {
+    // legacy powershell.exe seems to run faster than pwsh.exe so prefer it if available
+    String[] commands = new String[] { "powershell.exe", "pwsh.exe" };
+    IOException firstException = null;
+    for (String dir : dirs) {
+      File dirFile = new File(dir);
+      for (String command : commands) {
+        File commandFile = new File(dir, command);
+        if (commandFile.exists()) {
+          try {
+            return runCommands(guidsLength, Charset.forName("UTF-8"),
+                commandFile.toString(),
+                "-NoProfile",
+                "-EncodedCommand",
+                encodedCommand
+            );
+          } catch (IOException e) {
+            firstException = firstException == null ? e : firstException;
+          }
+        }
+      }
+    }
+    if (firstException != null) throw firstException;
+    else throw new IOException("no directories");
+  }
+
+  private static String[] windowsFallback(int guidsLength, Charset charset, String encodedCommand) {
+    File powerShellBase = new File("C:\\Program Files\\Powershell");
+    String[] powerShellDirs = powerShellBase.list();
+    if (powerShellDirs == null) powerShellDirs = new String[0];
+    String[] allPowerShellDirs = new String[powerShellDirs.length + 1];
+    // legacy powershell.exe seems to run faster than pwsh.exe so prefer it if available
+    allPowerShellDirs[0] = "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\";
```

Comment:
Maybe we could have used `%SystemRoot%\System32\WindowsPowerShell\v1.0\` here (`System.getenv("SystemRoot") + "\\System32\\…"`), if `SystemRoot` is set. It's mentioned in the links in https://github.com/dirs-dev/directories-jvm/issues/29#issuecomment-583900842 in particular.

Reply:
Done.